File: trunk/SFDataApplication/src/shadow/system/data/objects/SFMatrix3fData.java
package shadow.system.data.objects;

import shadow.math.SFMatrix3f;

public class SFMatrix3fData extends SFVectorData{
	
	private SFMatrix3f matrix;
	
	public SFMatrix3fData(SFMatrix3f matrix) {
		super(9);
		this.matrix=new SFMatrix3f();
		this.matrix.setArray(getFloatValues());
		this.matrix.set(matrix);
	}
	
	public SFMatrix3fData() {
		super(9);
		matrix=new SFMatrix3f(getFloatValues());
		this.matrix.set(SFMatrix3f.getIdentity());
	}

	public SFMatrix3f getMatrix3f(){
		return matrix;
	}
	
	@Override
	public SFVectorData copyDataObject() {
		return new SFMatrix3fData();
	}
}

File: SFDataApplicationTestC++/shadow/system/data/objects/SFMatrix3fData.cpp


#include "SFMatrix3fData.h"


namespace sf{
SFMatrix3fData::SFMatrix3fData(SFMatrix3f matrix) : SFVectorData(9){
	this->matrix.setArray(getFloatValues());
	this->matrix.set(&matrix);
}

SFMatrix3fData::SFMatrix3fData() : SFVectorData(9){
	matrix.setArray(getFloatValues());
    SFMatrix3f temp=SFMatrix3f::getIdentity();
	this->matrix.set(&(temp));
}

SFMatrix3f SFMatrix3fData::getMatrix3f(){
	return matrix;
}

SFVectorData* SFMatrix3fData::clone() {
	return new SFMatrix3fData();
}
}
